libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.kt

```diff
@@ -654,9 +670,144 @@ open class LoginActivity : AppCompatActivity(), OAuthWebviewHelperEvents {
 
     open fun onBioAuthClick(view: View?) = presentBiometric()
 
+    // region QR Code Login Via UI Bridge API Public Implementation
+
+    /**
+     * Automatically log in with a UI Bridge API login QR code.
+     * @param loginQrCodeContent The login QR code content.  This should be
+     * either a URL or URL query containing the UI Bridge API JSON parameter.
+     * The UI Bridge API JSON parameter should contain URL-encoded JSON with two
+     * values:
+     * - frontdoor_bridge_url
+     * - pkce_code_verifier
+     *
+     * If pkce_code_verifier is not specified then the user agent flow is used
+     * @return Boolean true if a log in attempt is possible using the provided QR
+     * code content, false otherwise
+     */
+    fun loginFromQrCode(
```

Comment:
All the new methods here should look _very_ familiar to @wmathurin.  I did apply quite a bit of comment and naming thought though to reflect my own journey in understanding how it all worked plus see if I could make the semantic match through all the reference documentation.  There's some Kotlin style added as well for extra credit.

Reply:
@JohnsonEricAtSalesforce @wmathurin Is there any kind of profile limitation to this login option? If not should we consider moving this to a standalone utility class that could be used by LoginActivity and native login consumers?  Perhaps native login support is a new story to itself, just trying to be forward thinking so we don't have to deprecate and move APIs around if possible.

Reply:
The bridge url is supposed to be loaded in a web view to get the tokens or the code - it is not a headless API. So it's another form of web view based login.

Reply:
Yes, but despite it being a web based login it is possible to do it without the customer ever seeing a webview.  You can put a button on a native login screen that has the same code as the `QrCodeEnabledLoginActivity` from the template app and if the user is pre authorized (a requirement for headless flow) they don't have to accept connected app policy.  This is obviously work well beyond the scope of this PR and would require a headless webview but it sounds possible, right?

Reply:
It's like the web view to get a code for the SP in the IDP-based login. You have to show it because user interaction might be needed. 
But you are right that you could launch it from the native login screen.

Reply:
Yeah, if the user is preauthorized, we could hide the web view. I guess we could let the mobile app know whether to show the web view or not by adding a field in the JSON we encode in the QR URL.